### PR TITLE
Remove mentions of kube-dns in kubeadm docs

### PR DESCRIPTION
**Pull request by rajansandeep**
Kubeadm is removing support for the deprecated kube-dns addon. This PR clears any mention of kube-dns from the kubeadm docs and also indicate that kube-dns is not supported from 1.21

Related to: https://github.com/kubernetes/kubeadm/issues/1943

### CODE REVIEW

**Comment by neolit123** [comment]
@rajansandeep 
as mentioned here, this should be for the `dev-1.21` branch of k/website:
https://github.com/kubernetes/kubernetes/pull/99646#issuecomment-789080543

File: content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md [hunk 1]
```diff
@@ -66,12 +66,10 @@ following steps:
 
 1. Installs a DNS server (CoreDNS) and the kube-proxy addon components via the API server.
    In Kubernetes version 1.11 and later CoreDNS is the default DNS server.
-   To install kube-dns instead of CoreDNS, the DNS addon has to be configured in the kubeadm `ClusterConfiguration`.
-   For more information about the configuration see the section `Using kubeadm init with a configuration file` below.
    Please note that although the DNS server is deployed, it will not be scheduled until CNI is installed.
 
    {{< warning >}}
-   kube-dns usage with kubeadm is deprecated as of v1.18 and will be removed in a future release.
+   kube-dns usage with kubeadm is deprecated as of v1.18 and will is removed in v1.21.
```
Comment (neolit123) [inline content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md:72]: leaving to sig docs to say if we can outline it like that.
i think the preference was to use version tags.

**Comment by neolit123** [comment]
tech review
/lgtm
/sig cluster-lifecycle

**Comment by mrbobbytables** [comment]
Can you please rebase this? There are quite a few commits in here

**Comment by rajansandeep** [comment]
I've messed up the branch too much.
I've opened a new PR instead. https://github.com/kubernetes/website/pull/26842

**Pull request closed without merging**